File: common/internal/src/main/java/dcp/mc/pstp/config/Builder.java
package dcp.mc.pstp.config;

import java.io.IOError;
import java.io.IOException;
import java.net.Proxy;

import dcp.mc.pstp.ModLoaderThings;
import dcp.mc.pstp.Utils;

import com.mojang.authlib.yggdrasil.YggdrasilAuthenticationService;
import org.jetbrains.annotations.ApiStatus.Internal;

@Internal
public enum Builder {
    INSTANCE;

    Builder() {
        var gson = Utils.gson;
        var folder = ModLoaderThings.getConfigDir().toFile();
        var file = ModLoaderThings.getConfigDir().resolve("projectsavethepets.json").toFile();
        var authenticationService = new YggdrasilAuthenticationService(Proxy.NO_PROXY);
        var profileRepository = authenticationService.createProfileRepository();
        var sessionService = authenticationService.createMinecraftSessionService();

        if (!folder.exists() && !folder.mkdirs()) {
            throw new IOError(new IOException("Failed to create configuration directory"));
        }


    }
}

File: common/src/main/kotlin/dcp/mc/pstp/config/Builder.kt
package dcp.mc.pstp.config

import java.nio.file.Path

object Builder {
    @JvmStatic
    val configDirectory: Path
        get() = throw AssertionError("This method should be overridden")
//
//    init {
//        val gson = GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES).setPrettyPrinting().create()
//        val folder = configDirectory.toFile()
//        val file = configDirectory.resolve("projectsavethepets.json").toFile()
//        val authenticationService = YggdrasilAuthenticationService(Proxy.NO_PROXY)
//        val profileRepository = authenticationService.createProfileRepository()
//        val sessionService = authenticationService.createMinecraftSessionService()
//
//        if (!folder.exists() && !folder.mkdirs()) {
//            throw FileNotFoundException("Failed to find and create the configuration directory!")
//        }
//
//
//    }
}